Create Terraform Script to integrate AWS ORGANIZATION (#1)
uptycslabs/terraform-aws-org-integration: README.md

# Terraform AWS IAM role module

> - This module allows you to create AWS IAM role with required policies and return role ARN which can be used for Uptycs service integration for each account in an organization or the accounts those have uptycs tags.

Reviewer: ```suggestion
- This module allows you to integrate AWS master account and any child account with Uptycs.
```

> - This terraform module will create a IAM Role in each account with the following policies attached:

Reviewer: ```suggestion
- This module will create an IAM Role in each account. The role has following policies attached:
```

  - policy/job-function/ViewOnlyAccess
  - policy/SecurityAudit
  - Custom read only policy access for required resources

## Create a <file.tf> file, paste below codes and modify as needed in locals.

```
locals {
  # Provide the directory for the creation of modules and providers for all accounts 
  filepath = ""
  # Provide the string to prefix the resources

>   resource_prefix = "cloudquery"

Reviewer: Change this to `integration_name` or something similar

| aws_account_id            | AWS organization's master account ID                            | `string` | `""`                | Yes      |
| external_id               | Role external ID provided by Uptycs                             | `string` | `""`                | Yes      |
| vpc_flowlogs_bucket_name  | Name of the S3 bucket that contains the VPC flow logs           | `string` | `""`                |          |
| cloudtrail_s3_bucket_name | Name of the S3 bucket which contains the CloudTrail data        | `string` | `""`                |          |
| kinesis_stream_name       | Name of the Kinesis stream configured to stream CloudTrail data | `string` | `""`                |          |
| tags                      | Tags to apply to the resources created by this module           | `map`    | `{}`                |          |

## Outputs

| Name           | Description                                     |
| ---------------- | ------------------------------------------------- |
| aws_parameters | AWS parameters (ExternalId and IntegrationName) |

## Notes:-

- The user should have `Administrators`  permission to the master account to create resources.

> - Every child account in the organization shoud have `OrganizationAccountAccessRole` role.

Reviewer: Add this link for more information
https://docs.aws.amazon.com/organizations/latest/userguide/orgs_manage_accounts_access.html
